### ISSUE TITLE
Throwing HealthCheckError in synchronous HealthIndicatorFunction does not work properly

### ISSUE THREAD

**Issue by andreialecu** [nestjs/terminus#2083]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

There is an issue with non-async health indicator functions due to the following: 

https://github.com/nestjs/terminus/blob/ea06b34b60d7710fd1e0227532960a678deaa355/lib/health-check/health-check-executor.service.ts#L63

If an error is thrown synchronously, then the health check will reply with `{"statusCode":500,"message":"Internal server error"}` and the server will log the error.

### Minimum reproduction code

```ts
@Get()
  @HealthCheck()
  async check() {
    const result = await this.health.check([
      () => throw new HealthCheckError('error', {}), // if changed to `async () => `, it works
    ])
  }
```

### Expected behavior

It should not depend on the function being async.

### Package version

9.1.3

### NestJS version

not relevant

### Node.js version

not relevant

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by andreialecu** [comment]
Fixed in #2084 (/cc @BrunnerLivio)

**Comment by BrunnerLivio** [comment]
@andreialecu Curious; why would you want to run a synchronous health check?

**Comment by andreialecu** [comment]
Because I have a specific case where I have a stream that sometimes randomly stops working, so I store a `lastMessageReceived` timestamp and check if nothing got in in the past 2 minutes, then fail the health check because of it.

I had to dig through the Terminus code to see why this wasn't working right.

If non async healthchecks should not be supported, the types could prevent a non-promise from being passed. But they specifically allow it:

https://github.com/nestjs/terminus/blob/8fdf979ae668124b06b9b05f6bc3937e9162bd2f/lib/health-indicator/health-indicator.ts#L8-L10

So this seemed like a bug to me.

**Comment by BrunnerLivio** [comment]
@andreialecu Alright makes sense to me! Let's go forward with your PR then :)

**Comment by BrunnerLivio** [comment]
Released with [9.1.4](https://github.com/nestjs/terminus/releases/tag/9.1.4) 🎉
